File: TypeScript/0344-反转字符串.ts
/**
 Do not return anything, modify s in-place instead.
 */
function reverseString(s: string[]): void {
  let l: number = -1,
    r: number = s.length;
  while (++l < --r) {
    [s[l], s[r]] = [s[r], s[l]];
  }
}

File: Java/0344-反转字符串.java
class Solution {
  public void reverseString(char[] s) {
    int l = -1, r = s.length;
    while (++l < --r) {
      s[l] ^= s[r];
      s[r] ^= s[l];
      s[l] ^= s[r];
    }
  }
}
